### PR TITLE
Add definedness conditions to some symbolic map rules

### PR DESCRIPTION
**Pull request by ana-pantilie**
This is needed for https://github.com/kframework/k/pull/1830

### CODE REVIEW

File: k-distribution/tests/regression-new/map-symbolic-tests-haskell/remove-9-spec.k.out [hunk 1]
```diff
@@ -25,6 +25,12 @@
   #Equals
     X:MyId in_keys ( MAP )
   }
+#And
+  {
+    false
+  #Equals
+    X:MyId in_keys ( MAP [ Y:MyId <- undef ] )
```
Comment (ttuegel) [inline k-distribution/tests/regression-new/map-symbolic-tests-haskell/remove-9-spec.k.out:32]: Can we add rules to simplify this more? (If so, we can do it in another pull request.)

**Review by ttuegel — APPROVED** [review]

File: k-distribution/tests/regression-new/map-symbolic-tests-haskell/remove-9-spec.k.out (hunk 1 above)
Comment (ana-pantilie) [reply k-distribution/tests/regression-new/map-symbolic-tests-haskell/remove-9-spec.k.out:32]: I noticed this as well. We should look into it as part of another issue I think.

**Review by dwightguth — APPROVED** [review]
